src/main.cpp

```diff
@@ -0,0 +1,252 @@
+#include <cmath>
+#include <cstdlib>
+#include <iostream>
+#include <math.h>
+#include <string>
+
+#include "ros/ros.h"
+#include "std_msgs/Bool.h"
+#include "std_msgs/String.h"
+
+#include <webots_ros/set_float.h>
+#include <webots_ros/set_int.h>
+#include <geometry_msgs/PointStamped.h>
+#include <geometry_msgs/Vector3.h>
+#include <sensor_msgs/Imu.h>
+
+
+std::string robotName {};
+std::string senderLocation {};
+std::string receiverLocation {};
+
+int gotRobotNameFlag{0};
+
+double yaw {0.0};
+double wheelRadius {0.165};
+double lengthBtnWheels {0.8};
+double maxSpeed {20};
+
+double currLocation[3] {};
+double g_setpoint[3] {100000.0,100000.0,100000.0};
+
+ros::NodeHandle* nh; //pointer
+
+webots_ros::set_float leftWheelSrv;
+webots_ros::set_float rightWheelSrv;
+
+ros::ServiceClient leftWheelClient;
+ros::ServiceClient rightWheelClient;
+
+ros::Publisher reachedSetpointPub;
+
+double findDistanceBetweenPoints(double point1[3], double point2[3])
+{
+    return sqrt((pow((point1[0]-point2[0]), 2)+pow((point1[2]-point2[2]),2)));
```

Comment:
usually before an equation, you have to comment and explain the analytical form the equation for better readability